src/nnue/evaluate_nnue.cpp

```diff
@@ -235,6 +235,7 @@ namespace Eval::NNUE {
             else
             {
                 sync_cout << "info string ERROR: failed to load eval file " << directory + eval_file << sync_endl;
+                eval_file_loaded.clear();
```

Comment:
It would be confusing that `eval_file_loaded` will not be empty when "SkipLoadingEval" is true.  But, if we clear `eval_file_loaded` when "SkipLoadingEval" is true, `verify_any_net_loaded()` will terminate the program.  Also `UseNNUE` should be `pure` when `learn()` is called.
How about to check
- a net is loaded or "SkipLoadingEval" is true, and
- `UseNNUE` is `pure`
in `verify_any_net_loaded()`?  We could also rename `verify_any_net_loaded()` according to the functionality.

Reply:
you're right, the current logic seems indeed slightly flawed. It stems from two facts:
1. load_eval returns true even when no eval was loaded
2. eval_file_loaded will not be empty even when using SkipLoadingEval.

How about we solve this by checking for "SkipLoadingEval" in the `init` and remove the check from `load_eval`, this removes the problem nr 1. For the problem nr 2 we add an additional check in the `verify_any_net_loaded` that if skiploadingeval then return;

I also like the proposed check for NNUE=pure because it's currently not enforced, but how about doing the check just before doing go_think in learn? Or maybe we should straight up set it to pure programmatically

Reply:
I speculatively pushed the changes proposed by me

Reply:
The change looks good to me.  Thank you.